### PR TITLE
Laravel 10

### LINKED ISSUE #5796: Upgrade to Laravel 10
Let's upgrade to Laravel 10 (codename: [X Gon' Give It To Ya](https://www.youtube.com/watch?v=fGx6K90TmCI))! 

🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈

## Leftover from Version 9
- Remove `api/server.php` [file](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/server.php) (no longer part of Laravel as of [version 9](https://github.com/laravel/laravel/commits/9.x/server.php))
- Update `api/.styleci.yml` file to [newest version](https://github.com/laravel/laravel/blob/10.x/.styleci.yml)

## Acceptance Criteria

- [ ] App is running on Laravel 10
- [ ] API works
- [ ] PHP unit tests work
- [ ] Login works
- [ ] Support tickets work
- [ ] CI works

## Reference

1. https://laravel-news.com/laravel-10
2. https://laravel.com/docs/10.x/upgrade

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #5796, #5789

## 👋 Introduction

This branch updates the app to use Laravel 10.

## 🕵️ Details

There were a few additional updates rolled into this as well due to the dependency graph.
- [fruitcake / laravel-cors](https://github.com/fruitcake/laravel-cors) was integrated into Laravel in version 9 and should no longer should be a dependency
- [fideloper / proxy](https://packagist.org/packages/fideloper/proxy) was integrated into Laravel in version 9 and should no longer should be a dependency
- [mll-lab / laravel-graphql-playground](https://github.com/mll-lab/laravel-graphql-playground) was deprecated in favour of [mll-lab / laravel-graphiql](https://github.com/mll-lab/laravel-graphiql)
- Upgrade to laravel/sanctum 3.2, doctrine/dbal 3.0, spatie/laravel-ignition 2.0, nunomaduro/collision 7.0, and phpunit/phpunit 10.0 (#5789) were part of the upgrade instructions

## 🧪 Testing

App is running on Laravel 10
API works
PHP unit tests work
Login works
Support tickets work - I'm not sure how to test this
CI works

### CODE REVIEW

File: infrastructure/conf/nginx-conf-local/default [hunk 1]
```diff
@@ -69,7 +69,7 @@ server {
         include fastcgi_params;
         fastcgi_param SCRIPT_FILENAME /home/site/wwwroot/api/public/index.php;
     }
-    location = /graphql-playground {
+    location = /graphiql {
```
Comment (mnigh) [inline infrastructure/conf/nginx-conf-local/default:72]: If switching to `graphiql`, there are a few references that should also be changed:
https://github.com/GCTC-NTGC/gc-digital-talent/blob/594d5cc6de52ea7ffc4d58799f438f703ab408f4/api/README.md?plain=1#L12
https://github.com/GCTC-NTGC/gc-digital-talent/blob/594d5cc6de52ea7ffc4d58799f438f703ab408f4/infrastructure/conf/xdebug.ini#L9

Comment (petertgiles) [reply infrastructure/conf/nginx-conf-local/default:72]: Good catch!
[update documentation for graphiql](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5869/commits/ff2e62ad247cda11218c8acfd3b4e2565ce9fb11)

**Comment by vd1992** [comment]
had this appear with api refresh

image

resolved by the same thing as last time, clearing out api/bootstrap/cache files before refreshing

**Comment by vd1992** [comment]
running phpunit, not sure why this test is marked as skip
unskipping it, and it seems to run fine, accidentally left in?

image

**Review by vd1992 — APPROVED** [review]
- refresh api ✅ 
 - migrate + seed ✅
 - artisan test ✅
 - run some e2e locally ✅
 - bounce around the app ✅

things seem good on my end, might as well un-skip that one test while you're at it

**Comment by vd1992** [comment]
may close #5873 too

**Comment by petertgiles** [comment]
I submitted a ticket!
image

**Pull request merged**